### PR TITLE
Remove emterpreter support.

### PR DESCRIPTION
**Pull request by sbc100**
Emterpreter has largerly been replaces by asyncify and is fastcomp only
so due for removing in the near future anyway.

The reason I'm pulling this forward ahead of fastcomp deprecation
is that it unblocks some other refactoring I'm working on.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ See docs/process.md for how version tagging works.
 
 Current Trunk
 -------------
+- Remove emterpreter and ``EMTERPRETIFY`` settings.  Emterpreter has largerly
+  been replaces by asyncify and is fastcomp only so due for removing in
```
Comment (kripken) [inline ChangeLog.md:21]: typo in `replaced`

File: tests/test_other.py [hunk 2]
```diff
@@ -5446,427 +5444,6 @@ def test_stat_fail_alongtheway(self):
...
-
-    print('check calling of emterpreted as well')
-    run_process([PYTHON, EMCC, 'src.c', '-s', 'EMTERPRETIFY=1', '-s', 'EMTERPRETIFY_ASYNC=1'])
-    self.assertContained('Success', run_js('a.out.js'))
-
-    print('check for invalid EM_ASM usage')
-    run_process([PYTHON, EMCC, 'src.c', '-s', 'EMTERPRETIFY=1', '-s', 'EMTERPRETIFY_ASYNC=1', '-s', 'EMTERPRETIFY_BLACKLIST=["_not_emterpreted"]', '-DBAD_EM_ASM'])
-    self.assertContained('cannot have an EM_ASM on the stack when emterpreter pauses/resumes', run_js('a.out.js', stderr=STDOUT, assert_returncode=None))
-
-  def test_link_with_a_static(self):
```
Comment (kripken) [inline tests/test_other.py:5834]: why was this test removed? looks like it might be accidental.

File: tests/test_core.py [hunk 3]
```diff
@@ -7775,14 +7719,14 @@ def test_async(self, emterpretify=False):
   printf("Hello");
   emscripten_async_call(f, &i, 1);
   printf("World");
-  emscripten_%s(100);
-  printf("%%d\n", i);
+  emscripten_slepp(100);
```
Comment (kripken) [inline tests/test_core.py:7722]: typo, should be `sleep`

**Comment by sbc100** [comment]
I did end up having to disable all of the async browser tests.. which I guess makes sense.

Landing this will allow us to remove the swappable module support which in turn helps me to land:
https://github.com/emscripten-core/emscripten/pull/11048

**Pull request merged**